src/components/Login.tsx

```diff
@@ -1,47 +1,62 @@
 import { useRef, useState } from "react";
-import { checkValidEmail } from "./utils/validate";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  updateProfile,
+} from "firebase/auth";
+import { Link, useNavigate } from "react-router-dom";
+import { checkValidEmail } from "../utils/validate";
 import HeaderComponent from "./ui/HeaderComponent";
-import { Link } from "react-router-dom";
-import { auth } from "./utils/firebase";
-import { createUserWithEmailAndPassword, signInWithEmailAndPassword } from "firebase/auth";
+import { auth } from "../utils/firebase";
 
 const Login: React.FC = () => {
+  type Details = {
+    email: string;
+    password: string;
+  };
```

Comment:
types can be defined outside the Login(react component )of a particular file

Reply:
declared outside component